server/consumer.go

```diff
@@ -386,7 +378,7 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 			} else {
 				// If we are a push mode and not active and the only difference
 				// is deliver subject then update and return.
-				if configsEqualSansDelivery(ocfg, *config) && (config.allowNoInterest || eo.hasNoLocalInterest()) {
+				if configsEqualSansDelivery(ocfg, *config) && eo.hasNoLocalInterest() {
```

Comment:
Should make this SC aware like we did yesterday for the consumer fix.

Reply:
This code is rebased from main.. If you made a change here, and is not yet merged, then it will be added during the merge. But maybe I am misunderstanding what you are asking.

Reply:
ok so maybe this is something I need to change on my side for GW interest checks etc.